interpreter/runtimecost.go

```diff
@@ -170,19 +188,60 @@ func (c CostTracker) actualSize(value ref.Val) uint64 {
 	return 1
 }
 
+type stackVal struct {
+	Val ref.Val
+	ID  int64
+}
+
 // refValStack keeps track of values of the stack for cost calculation purposes
-type refValStack []ref.Val
+type refValStack []stackVal
 
-func (s *refValStack) push(value ref.Val) {
+func (s *refValStack) push(value stackVal) {
 	*s = append(*s, value)
 }
 
-func (s *refValStack) pop(count int) ([]ref.Val, bool) {
-	if len(*s) < count {
-		return nil, false
+// drop1 removes the top item from the stack.
+func (s *refValStack) drop1() {
```

Comment:
If the end of the internal storage is considered the top, you can leave this as `pop`. I'm not sure why you wouldn't just use the IDs always though if you have them.

Reply:
Yeah, this section of code is not great.  This was the only remaining case where I didn't have easy access to an ID when popping.  I've decided to just leave the value on the stack and pop it, that way all pops are guaranteed safe.